src/libraries/SwapMath.sol

```diff
@@ -82,7 +82,7 @@ library SwapMath {
                 // we didn't reach the target, so take the remainder of the maximum input as fee
                 feeAmount = uint256(-amountRemaining) - amountIn;
             } else {
-                feeAmount = FullMath.mulDivRoundingUp(amountIn, feePips, 1e6 - feePips);
+                feeAmount = feePips == 1e6 ? amountIn : FullMath.mulDivRoundingUp(amountIn, feePips, 1e6 - feePips);
```

Comment:
Consider an exact output swap with fixed input args except `feePips`.
A reasonable expectation would be

> The fee amount monotonically increases as `feePips` increases.

On Ln 85, `amountIn` is determined by `sqrtRatioCurrentX96`, `sqrtRatioTargetX96`, `liquidity` and `amountRemaining`. So given a fixed `amountIn`, `feeAmount` is only a function of `feePips`. Under the current implementation:

- If `feePips == 0`, `feeAmount == 0`.
- If `feePips == 5e5`, `feeAmount == amountIn`.
- If `feePips == 1e6 - 1`, `feeAmount == 999999 * amountIn`.
- If `feePips == 1e6`, `feeAmount == amountIn`.

It breaks the above expectation. It also enables the change to pool price when the fee is 100%, which is disabled for the `exactIn` case.

Reply:
Thanks @shuhuiluo we disabled the case for 100% swap fee on exactOut